plugins/woocommerce/src/StoreApi/Utilities/OrderController.php

```diff
@@ -391,7 +391,7 @@ protected function validate_address_fields( \WC_Order $order, $address_type, \WP
 		$address = array_merge( $address, $additional_fields );
 
 		foreach ( $current_locale as $address_field_key => $address_field ) {
-			if ( empty( $address[ $address_field_key ] ) && $address_field['required'] ) {
+			if ( ! isset( $address[ $address_field_key ] ) && $address_field['required'] ) {
```

Comment:
I think a blank space string or empty string will bypass this required field check, also this might throw a warning if the key doesn't exist or is of an incorrect type. A mix of array_key_exists, is_string and trim($string) !== '' might do the trick 👍

Reply:
Thanks Paulo, I had similar thoughts but I decided against the extra checks because the value comes from client side where the input is trimmed and validated for empty strings if it is required. Also, this will need to work with checkbox type inputs soon too, so this if statement may get quite complex. But thinking about it some more, it's not guaranteed the input comes from the client, could be a direct request potentially from headless or somewhere else. Either way it's probably a good idea to harden this a bit, you're right